File: oop/advertiser.cpp
#ifndef __YEKTANET_ADVERTISER__
#define __YEKTANET_ADVERTISER__

#include "base_model.cpp"

class Advertiser : public BaseAdvertising {
  private:
    string name;
    static int totalClicks;
  public:
    Advertiser(int _id, string _name): BaseAdvertising(_id) {
      this->name = _name;
    }

    string getName(){
      return this->name;
    }
    void setName(string x){
      this->name = x;
    }
    
    void incClicks(){
      this->BaseAdvertising::incClicks();
      Advertiser::totalClicks++;
    }
    static int getTotalClicks() {
      return Advertiser::totalClicks;
    }

    void describeMe() {
      cout << "I am an Advertiser\n";
    }

    static void help() {
      cout << "class e advertiser ye seri method dare\n";
    }
};

int Advertiser::totalClicks;

#endif

File: python/advertiser.py
from base_model import *

class Advertiser(BaseAdvertising):
  __totalClicks = 0
  def __init__(self, id, name):
    super().__init__(id)
    self.__name = name

  def getName(self):
    return self.__name
  def setName(self, x):
    self.__name = x
    
  def incClicks(self):
    super().incClicks()
    self.__class__.__totalClicks += 1
    
  @classmethod
  def getTotalClicks(cls):
    return cls.__totalClicks

  def describeMe(self):
    print("I am an instance of the baseAdvertising class")

  @staticmethod  
  def help():
    print("class e advertiser ye seri method dare")
